DataFormats/L1TParticleFlow/interface/PFTau.h

```diff
@@ -72,11 +74,24 @@ namespace l1t {
     }
     bool passTightPF() const { return fullIso_ < PFTAU_PF_TIGHT_CUT; }
 
+    void set_encodedTau(std::array<uint64_t, 2> encodedTau) { encodedTau_ = encodedTau; }
+    std::array<uint64_t, 2> encodedTau() const { return encodedTau_; }
+
+    //l1gt::Tau getHWTauGT() const { return l1ct::Tau::unpack_gt(encodedTau_); }
+
+    //Return the l1gt Tau object for now
+    void set_TauGT(l1gt::Tau TauGT) { TauGT_ = TauGT; }
+
+    l1gt::Tau getTauGT() const { return TauGT_; }
+
   private:
+    float NNValues_[80];
     float iso_;
     float fullIso_;
     int id_;
     float dxy_;
+    std::array<uint64_t, 2> encodedTau_;
+    l1gt::Tau TauGT_;
```

Comment:
I don't think you can write out to root file objects that contain `ap_*` data members like `l1gt::Tau` - I just tried this and it failed.  Does it work for you?

To fix this, and to be consistent with what's been implemented for the jets, you can store only the `encodedTau_` and implement a `l1gt::Tau::unpack` for use with `getTauGT()` (done [here](https://github.com/cms-l1t-offline/cmssw/pull/1088/files#diff-b728d605f568ab3466237040728947ab6e564a9e94d237d40803fc1cd8fe411dR54) for the jets).